supabase/migrations/20240316051441_tables_add_threads_and_messages..sql

```diff
@@ -0,0 +1,39 @@
+CREATE TABLE IF NOT EXISTS threads (
+  id uuid NOT NULL PRIMARY KEY,
+  service_provider_id uuid NOT NULL,
+  customer_id uuid NOT NULL,
+  created_at timestamptz NOT NULL DEFAULT NOW(),
+  updated_at timestamptz NOT NULL DEFAULT NOW(),
+
+  FOREIGN KEY (service_provider_id) REFERENCES service_providers(user_id),
+  FOREIGN KEY (customer_id) REFERENCES customers(user_id)
+);
```

Comment:
For now, we'll bake in the assumption that chats will only be 1-on-1.

We can work in admin / moderator access via the `service_role` role or adding an admin role to the `auth.users` row. We'll leave that for when the time comes.

Reply:
Also, I need to think about how I want to access this data. Probably something like `get all threads belonging to me as a user`.

Reply:
Might make sense to add a `latest_thread_id` field here as well, if we want to display the latest message like this screenshot from chatscope:
![image](https://github.com/codesydney/marketplace-app-for-good/assets/8443215/dd9d59a2-d300-4ae3-8e72-ed1f6742c7eb)

Could use a join query for the threads / conversations panel